test/beforeHeadersSend.test.js

```diff
@@ -29,172 +32,241 @@ let setMock;
+                setSpendFlag(url.host, true);
+                let redeemHdrs = beforeSendHeaders(details, url);
+                expect(redeemHdrs.cancel).toBeFalsy();
+                expect(redeemHdrs.requestHeaders).toBeFalsy();
+            });
+            test(`max spend has been reached, config.id = ${config.id}`, () => {
+                setSpendFlag(url.host, true);
+                setSpentHosts(url.host, 31);
+                switch (config.id) {
+                case 1:
+                    workflow.__set__("SPEND_MAX", 3);
+                    break
+                case 2:
+                    workflow.__set__("SPEND_MAX", 0);
+                    break
+                default:
+                    throw Error(`Unhandled config.id, ${config.id}`)
+                }
+                let redeemHdrs = beforeSendHeaders(details, url);
+                expect(redeemHdrs.cancel).toBeFalsy();
+                switch (config.id) {
+                    case 1:
+                        expect(redeemHdrs.requestHeaders).toBeFalsy();
+                        break
+                    case 2:
+                        // hCaptcha has no SPEND_MAX
+                        expect(redeemHdrs.requestHeaders).toBeTruthy();
+                        break
+                    default:
+                        throw Error(`Unhandled config.id value => ${config.id}`)
+                }
+            });
+            test(`spend has been attempted for url, config.id = ${config.id}`, () => {
+                setSpendFlag(url.host, true);
+                setSpentHosts(url.host, 0);
+                setSpentUrl(url.href, true);
+                setReedemMethod(config.id);
+                let redeemHdrs = beforeSendHeaders(details, url);
+                expect(redeemHdrs.cancel).toBeFalsy();
+                switch (config.id) {
+                    case 1:
+                        expect(redeemHdrs.requestHeaders).toBeFalsy();
+                        break
+                    case 2:
+                        // hCaptcha will always spend on its URLS
+                        expect(redeemHdrs.requestHeaders).toBeTruthy();
+                        break
+                    default:
+                        throw Error(`Unhandled config.id value => ${config.id}`)
+                }
+
+            });
+            test(`redemption method is not reload, config.id = ${config.id}`, () => {
+                setSpendFlag(url.host, true);
+                setSpentHosts(url.host, 0);
+                setSpentUrl(url.href, false);
+                workflow.__set__("REDEEM_METHOD", "invalid");
+                let redeemHdrs = beforeSendHeaders(details, url);
+                expect(redeemHdrs.cancel).toBeFalsy();
+                expect(redeemHdrs.requestHeaders).toBeFalsy();
+            });
+            test(`no token to spend, config.id = ${config.id}`, () => {
+                localStorage = {
+                    "cf-bypass-tokens": `{}`,
```

Comment:
I know this is old code, but I guess this should actually be `bypass-tokens-${config.id}` and `bypass-tokens-count-${config.id}`.